### ISSUE TITLE
parents file

### ISSUE THREAD

**Issue by adriludwig** [dblhlx/PyBSASeq#9]
Hi!
Sorry for my dumb questions, but I'm really interested in using your program and I don't have experience with bulk segregation analysis. I understood that the parent genome is not needed in the current version of the program, but as I have, I suppose it would be better to use it. In this case, the parents.tsv is the SNP calling of both parents using one of the parents as a reference? Another question, I see the figure generated are "Number of SVs" and and not "Number of SNPs" as in the papers. Am I using the correct version of the program?

Thanks very much

**Comment by dblhlx** [comment]
Hi Adriana,
No question is a dumb question. It is that the author hasn't made it clear enough if a reader has any question.
Yes, the genome sequences of the parents are not needed, but it is nice to have them. Theoretically, the SNP/InDel sets obtained via comparing the parents should be the same as those obtained via comparing the bulks. But they are always different in some extend.
If the genomes of both the bulks and the parents are sequenced, I will first perform SNP/InDel calling using only the sequences of the parents to obtained a SNP/InDel set, and then do the same for the bulks to obtain another SNP/InDel set. The common SNPs/InDels of these two sets are used for BSA-Seq data analysis in this case. Although good enough results can be obtained using only the bulk sequences, the results are less noisy and the detection power is a little higher when the sequences of both parents and bulks are used.
You have the correct version of the program. The script can handle both SNPs and InDels. SV (structural variant) is a general term for both SNPs and InDels.
Please let me know if you encounter any issue.

**Comment by adriludwig** [comment]
Thanks very much for your kind and quick answer. I do believe your program can be very useful for my experiment. I tested now only bulks mapped against a non-parent reference, option RIL. I haven't done any complex filtering in the vcf file. I eliminated only the multiallelic SNPs. But everything is above the threshold line in the second plot (attached). Do you have any clues why? Anyway, I'm going to proceed the analysis including the parents: 1- mapping parents and bulks against the available parent genome; 2 - call variants for the 2 parents and the 2 bulks separately using the parent genome as ref; 3- I suppose I need to filter common variants before; 4- applying pyBSAseq using filtered parents and bulks tsv files? 
<img width="930" alt="Screenshot 2024-01-21 at 16 08 15" src="https://github.com/dblhlx/PyBSASeq/assets/49766048/24116005-70fd-4885-bd07-e23547e9049b">

Thanks so much again!

**Comment by dblhlx** [comment]
We can estimate confidence intervals/thresholds at either the SV (SNP/InDel) level (the magenta curves in panels c, and d) or at the genomic region level (the red lines in panels b, c, and d). Realistically, we can identify genomic regions, not SNPs, associated with a trait or interest. So it's more proper to estimate confidence intervals/thresholds at the genomic region level. The sSV/totalSV ratio is a parameter at the genomic region level, and its threshold can be estimated only at the genomic region level. BSA-Seq has much higher statistical power when we estimate confidence intervals/thresholds considering a sliding window can contains hundreds/thousands of SVs. This is why all peaks in panel b are above the threshold.  Thresholds/confidence intervals can also be estimated in panels c and d, and the results are very similar to that in panel b. You can find more details in my new manuscript (https://www.biorxiv.org/content/10.1101/2023.03.12.532308v2).
There is a bug in my previous code dealing with panel d. Please download the updated code and rerun the program.
The program can filter out multiallelic SNPs and identify the common SNP set if both the parents.tsv and bulks.tsv files are provided. So You don't need to do anything extra.
Good luck!

**Comment by adriludwig** [comment]
Hi! Thanks again for your explanations! I realized that the program would do the filtering steps! I'm doing some tests with different window sizes and using the parents (I can already see it improved the plots). I think one of my problems is that I'm merging bam files from individuals to get the 2 bulks without considering the depth coverage that varies a lot among the samples (we haven't done pool sequencing). I'm quite sure this is the wrong way to do it since I have very distinct contributions of individuals in the bulks.  In parallel I'm doing some GWAS since I have the individual sequencing, but I would like to use both approaches. I'll take a look on your new paper. Thanks!

**Comment by dblhlx** [comment]
It is imperative that each individual in a bulk contributes the same amount of sequencing reads in BSA-Seq. You can sample the same amount of sequencing reads from each individual fastq file using the seqkit tool (https://github.com/shenwei356/seqkit).

**Comment by adriludwig** [comment]
Thanks again! I downsized the bam files to normalize them. Still, in the plot B, everything is above the threshould, so I don't know what else I should do. The parents and offspring were mapped against a third genome. I used -p RIL -s 200000,10000
Would the mapping against one of the parents improve the analysis.


<img width="1288" alt="Screenshot 2024-01-29 at 10 24 20" src="https://github.com/dblhlx/PyBSASeq/assets/49766048/c1b983db-79a5-4d7c-b68b-c8e68b8ff226">

**Comment by dblhlx** [comment]
It is okay that all peaks are above the threshold. Each significant peak represents a gene that contributes to the phenotype, and SNPs within and around the peaks will be enriched/depleted by bulking. SNPs within the gene will be enriched/depleted the most while SNPs surrounding the gene will be a little less enriched/depleted due to recombination. It is possible that all sSV/totalSV ratios of a chromosome are above the threshold, provided that many genes in the chromosome are involved in controlling the phenotype and/or recombination frequency is low.

Another reason that everything is above the threshold is that the p-value for identifying sSNP in the simulated dataset is set to 0.01. For the significant structural variant method (panel b), we need to identify significant structural variants (sSV) in both the real dataset and the simulated dataset. The default p-value for both situations is 0.01. You can set the p-value for identifying sSV in the simulated dataset a little higher (e.g., 0.05 or 0.1) to increase the threshold. But you will has higher chance to get more false-negatives if you do so. The options should be something as below:
-p RIL -s 200000,10000 -v 0.01,0.05

**Comment by adriludwig** [comment]
Thanks! I'll try to change the p-value and see what I get! You've been very helpful!
